builder/builder.py

```diff
@@ -0,0 +1,138 @@
+import click
+import os
+import sys
+import logging
+
+TIMESTAMP_FORMAT = '%Y-%m-%dT%H:%M:%S'
+SHORT_TIMESTAMP_FORMAT = '%Y-%m-%d'
+
+logger:object = logging.getLogger('Artifact Builder ===>')
+logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
+class builder:
+    def __init__(self, *args, **kwargs) -> None:
+        print('Builder started')
+
+    def find_executable(self, executable, path=None):
+        """Tries to find 'executable' in the directories listed in 'path'.
+
+        A string listing directories separated by 'os.pathsep'; defaults to
+        os.environ['PATH'].  Returns the complete filename or None if not found.
+        """
+        if path is None:
+            path = os.environ.get('PATH', os.defpath)
+
+        paths = path.split(os.pathsep)
+        base, ext = os.path.splitext(executable)
+
+        if (sys.platform == 'win32' or os.name == 'os2') and (ext != '.exe'):
+            executable = executable + '.exe'
+
+        if not os.path.isfile(executable):
+            for p in paths:
+                f = os.path.join(p, executable)
+                if os.path.isfile(f):
+                    # the file exists, we have a shot at spawn working
+                    return f
+            return None
+        else:
+            return executable
+
+    def clean_up(self, project_dir:str):
+        target_dir = os.path.join(project_dir, 'target')
+        build_dir = os.path.join(project_dir, 'build')
+        os.system('rm -Rf {}'.format(target_dir))
+        os.system('rm -Rf {}'.format(build_dir))
+
+@click.command()
+@click.option('-d', '--project-dir',
+              help='password for git', required=True, type=str)
```

Comment:
Yes, it is intentional 
When Jenkins checks out a branch, it creates some working directory like cnmResponse127xxaasdf and try to clean it up when the task is done.  I have learned in the hard way when doing Jenkins pipeline for other Java Lambdas.   Files created/touched within the docker belongs to the users within docker. Hence, they can not be cleaned by Jenkins. The code is just trying to be nice by giving jenkins permission to delete everything within its working directory - instead of leaving many debris on the disk.   This all happened within Jenkins task so should be no security concern.